source/common/stats/thread_local_store.cc

```diff
@@ -283,6 +310,27 @@ void ThreadLocalStoreImpl::clearScopeFromCaches(uint64_t scope_id,
   }
 }
 
+void ThreadLocalStoreImpl::clearHistogramsFromCaches() {
+  // If we are shutting down we no longer perform cache flushes as workers may be shutting down
+  // at the same time.
+  if (!shutting_down_) {
+    // Capture all the pending histograms in a local, clearing the list held in
+    // this. Once this occurs, if a new histogram is deleted, a new post will be
+    // required.
+    auto histograms = new std::vector<uint64_t>();
```

Comment:
I know the ownership is a little wonky with how runOnAllThreads works, but could you make this a unique_ptr, have complete_cb (2nd closure) take the unique_ptr by value, and have cb (1st closure) take `*histograms` by `&`?

Reply:
I think not; but it can work as a shared_ptr.